app/src/main/java/com/example/faunaapp/view_model/AddEntryFragmentViewModel.java

```diff
@@ -22,36 +36,61 @@ public AddEntryFragmentViewModel() {
         date = new MutableLiveData<>();
         time = new MutableLiveData<>();
         error = new MutableLiveData<>();
+        executorService = Executors.newFixedThreadPool(2);
+        ClientComponent component = DaggerClientComponent.create();
+        client = component.getClient();
+        entryModel = new AddEntryModel(client);
     }
 
-    public void chooseDate(String date){
+    public void chooseDate(String date) {
         this.date.postValue(constantDateText + "\n" + date);
     }
-    public void chooseTime(String time){
-        this.time.postValue(constantTimeText +"\n" + time);
+
+    public void chooseTime(String time) {
+        this.time.postValue(constantTimeText + "\n" + time);
     }
 
-    public void submitTheEntry(Entry entry){
+    public void submitTheEntry(Entry entry) {
+        String errorMessage = getErrorIfExists(entry);
+        if(!errorMessage.equals(""))
+        {
+            error.postValue(errorMessage);
+            return;
+        }
+        error.postValue("");
+        entry.setDate(entry.getDate().substring(entry.getDate().lastIndexOf("\n") + 1));
```

Comment:
Try and avoid using indexing when you can, it is super hard to work with and therefore super error prone. I do not understand why it is necessary here for example. Why do you want a newline?

Reply:
I understand what you mean and it is a dirty code. I will gonna change it